### PR TITLE
Modify some syscalls

### PR DESCRIPTION
**Pull request by Miochyann**
- Add syscalls：sys_geteuid，sys_getegid.
   - Modify syscalls：pipe.read(). Previously, when I tried to read data, the writing end was not closed, but by then, the reading end had completed the read but did not return as expected.
   - Write a doc and script about udpserver
   - Modify the error return errno in non-blocking mode for TCP connections.

### CODE REVIEW

File: apps/c/udpserver/features.txt [hunk 1]
```diff
@@ -1,3 +1,4 @@
 alloc
 paging
 net
+fs
```
Comment (coolyjg) [inline apps/c/udpserver/features.txt:4]: Why does this app need `fs` feature?

File: modules/axnet/src/smoltcp_impl/tcp.rs [hunk 2]
```diff
@@ -11,7 +11,7 @@ use core::cell::UnsafeCell;
 use core::net::SocketAddr;
 use core::sync::atomic::{AtomicBool, AtomicU8, Ordering};
 
-use axerrno::{ax_err, ax_err_type, AxError, AxResult};
+use axerrno::{ax_err, ax_err_type, AxError, AxResult, LinuxError};
```
Comment (coolyjg) [inline modules/axnet/src/smoltcp_impl/tcp.rs:14]: `LinuxError` is not used.

File: modules/axnet/src/smoltcp_impl/udp.rs [hunk 3]
```diff
@@ -217,7 +217,9 @@ impl UdpSocket {
 
     fn send_impl(&self, buf: &[u8], remote_endpoint: IpEndpoint) -> AxResult<usize> {
         if self.local_addr.read().is_none() {
-            return ax_err!(NotConnected, "socket send() failed");
+            let res = SocketAddr::V4(SocketAddrV4::new(Ipv4Addr::new(0, 0, 0, 0), 0));
+            self.bind(res)?;
+            //return ax_err!(NotConnected, "socket send() failed");
```
Comment (coolyjg) [inline modules/axnet/src/smoltcp_impl/udp.rs:222]: remove this line.

File: modules/axnet/src/smoltcp_impl/udp.rs [hunk 4]
```diff
@@ -288,7 +290,8 @@ impl Drop for UdpSocket {
 }
 
 fn get_ephemeral_port() -> AxResult<u16> {
-    const PORT_START: u16 = 0xc000;
+    //const PORT_START: u16 = 0xc000;
```
Comment (coolyjg) [inline modules/axnet/src/smoltcp_impl/udp.rs:293]: remove this line.

File: api/ruxos_posix_api/src/imp/fs.rs [hunk 5]
```diff
@@ -280,7 +280,21 @@ pub unsafe fn sys_fstat(fd: c_int, kst: *mut core::ffi::c_void) -> c_int {
         }
     })
 }
+/// Obtain a valid user ID for the process
+///
+/// return 1000 if success.
```
[the remark this answers is not in the document]
Comment (Miochyann) [reply api/ruxos_posix_api/src/imp/fs.rs:285]: We have set `st_uid = 1000` and` st_gid = 1000` in `fs.rs`， so I return 1000.

File: api/ruxos_posix_api/src/imp/net.rs [hunk 6]
```diff
@@ -251,6 +252,11 @@ pub fn sys_socket(domain: c_int, socktype: c_int, protocol: c_int) -> c_int {
             | (ctypes::AF_INET, ctypes::SOCK_DGRAM, 0) => {
                 Socket::Udp(Mutex::new(UdpSocket::new())).add_to_fd_table()
             }
+            (ctypes::AF_INET, _SOCK_DGRAM_NONBLOKCING, ctypes::IPPROTO_TCP) => {
```
[the remark this answers is not in the document]
Comment (Miochyann) [reply api/ruxos_posix_api/src/imp/net.rs:255]: -  When I try to match directly with` ctypes::SOCK_STREAM | ctypes::SOCK_NONBLOCK`, the matching fails，so i use `_SOCK_STREAM_NONBLOK`
warning: unreachable pattern
```
   --> api/ruxos_posix_api/src/imp/net.rs:256:31
    |
256 |             (ctypes::AF_INET, ctypes::SOCK_STREAM | ctypes::SOCK_NONBLOCK, ctypes::IPPROTO_TCP) => {
    |                               ^^^^^^^^^^^^^^^^^^^
    |
    = note: `#[warn(unreachable_patterns)]` on by default
```

- I'm sorry for submitting the wrong code. The correct code is `_SOCK_STREAM_NONBLOK: u32 = ctypes::SOCK_STREAM | ctypes::SOCK_NONBLOCK;`

**Pull request closed without merging**